app/controllers/consultees_controller.rb

```diff
@@ -3,12 +3,38 @@
 class ConsulteesController < AuthenticationController
   before_action :set_planning_application
 
+  def edit
+    @consultee = planning_application.consultees.find(params[:id])
+
+    respond_to do |format|
+      format.html
+    end
+  end
+
   def create
     consultee = planning_application.consultees.new(consultee_params)
     consultee = planning_application.consultees.new if consultee.save
     render(json: { partial: consultees_partial(consultee) })
   end
 
+  def update
+    @consultee = planning_application.consultees.find(params[:id])
+    if @consultee.update(consultee_params)
+      if (consultation_summary = planning_application.consultation_summary)
+        redirect_to edit_planning_application_assessment_detail_path(planning_application, consultation_summary)
+      else
+        # it's possible that the list of consultees is being edited before the
+        # consultation summary has been saved and so we can't link to the edit path
+        redirect_to new_planning_application_assessment_detail_path(
+          planning_application,
+          category: "consultation_summary"
+        )
+      end
```

Comment:
I'd extract out the redirect path into a private helper method to make the logic clearer, e.g.

``` ruby
def update
  @consultee = planning_application.consultees.find(params[:id])

  if @consultee.update(consultee_params)
    redirect_to after_update_consultees_path
  else
    render :edit
  end
end

private

  def after_update_consultees_path
    if consultation_summary = planning_application.consultation_summary
      edit_planning_application_assessment_detail_path(planning_application, consultation_summary)
    else
      # it's possible that the list of consultees is being edited before the
      # consultation summary has been saved and so we can't link to the edit path
      new_planning_application_assessment_detail_path(planning_application, category: "consultation_summary")
    end
  end
```

Also are we not doing 'update successful' type flash messages?

Reply:
> Also are we not doing 'update successful' type flash messages?
yes but I keep forgetting them 🤦🏻